org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/TestParticipantFactory.java

```diff
@@ -0,0 +1,42 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Microsoft Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Microsoft Corporation - initial API and implementation
+ *******************************************************************************/
+
+package org.eclipse.jdt.ls.core.internal;
+
+import java.util.List;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IExecutableExtensionFactory;
+
+public class TestParticipantFactory implements IExecutableExtensionFactory {
+
+	private static class TestParticipant implements IParticipant {
+
+		public static TestParticipant getInstance() {
+			return new TestParticipant();
+		}
+
+		private int state = 0;
+
+		@Override
+		public Object executeCommand(String commandId, List<Object> arguments) {
+			if (commandId.equals("testcommand.throwexception")) {

```

Comment:
`"testcommand.throwexception".equals(commandId)`

always put the string constant 1st, to avoid potential NPEs in comparisons